magma/primitives/when.py

```diff
@@ -165,7 +165,14 @@ def _generic_add(
         port_name = f"{name_prefix}{next(name_counter)}"
         self._add_port(port_name, type_qualifier(type(value).undirected_t))
         with no_when():
-            wire(getattr(self, port_name), value)
+            port = getattr(self, port_name)
+            if value.is_input() and isinstance(value, Enable):
+                value.wire(
+                    port,
+                    driven_implicitly_by_when=value.driven_implicitly_by_when
+                )
+            else:
+                wire(port, value)
```

Comment:
```suggestion
            port = getattr(self, port_name)
            wire_kwargs = {}
            if value.is_input() and isinstance(value, Enable):
                wire_kwargs["driven_implicitly_by_when"] = value.driven_implicitly_by_when
            value.wire(port, **wire_kwargs)
```

believe this is equiv.

Reply:
This doesn't work for two reasons:
(1) We need to use `wire(...)` in the general case since this method is used for both inputs/outputs
(2) The driven_implicitly_by_when kwarg is only provided by the Enable wire method

We could: add the implicitly_by_when kwarg to all the wire locations, but this seems like a big change for little benefit.